src/error.jl

```diff
@@ -27,16 +25,62 @@ end
 function gdaljl_errorhandler(class::CPLErr, errno::Cint, errmsg::Cstring)
     # function signature needs to match the one in __init__, and the signature
     # of the callback for a custom error handler in the GDAL docs
-    if class in throw_class
+    if class === CE_Failure
         throw(GDALError(class, errno, unsafe_string(errmsg)))
     end
     return C_NULL
 end
 
-"Throw an error if a pointer is null and GDAL reports an error"
-function failsafe(ptr)
-    if ptr == C_NULL && cplgetlasterrortype() in throw_class
+"Check the last error type and throw a GDALError if it is a failure"
+function maybe_throw()
+    if cplgetlasterrortype() === CE_Failure
         throw(GDALError())
     end
-    ptr
+    nothing
+end
+
+"""
+Handle anything returned from GDAL
+
+When values are returned from GDAL, always check if there was a failure using `maybe_throw`.
+
+Depending on the return type, we do extra work.
+"""
+function aftercare(x)
+    maybe_throw()
+    x
+end
+
+"For string pointers, load them to String, and free them if we should."
+function aftercare(ptr::Cstring, free::Bool)
+    maybe_throw()
+    if ptr == C_NULL
+        return nothing
+    else
+        s = unsafe_string(ptr)
+        free && vsifree(convert(Ptr{Cvoid}, ptr))
```

Comment:
Freeing memory utils could also be placed in a `memory.jl` file? `errors.jl` is just not the place I would expect them.

Reply:
True that `error.jl` is not only about errors anymore. But it's less that 100 lines and I'd like to keep the `aftercare`s together. Not sure of a better name for `error.jl` though, since it's mostly that.